### PR TITLE
add expiry control to invoice command

### LINKED ISSUE #435: [invoice] add option to control the expiration time
allow overriding the default expiry

### PR DESCRIPTION
**Pull request by niteshbalusu11**
Signed-off-by: Nitesh Balusu <niteshbalusu@icloud.com>
Closes #435 

Added expiry control to invoice command.

Files changed:
offchain/create_invoice.js
offchain/sign_payment_request.js

### CODE REVIEW

File: bos [hunk 1]
```diff
@@ -1062,6 +1062,7 @@ prog
   .help('--virtual invoices cannot be used with payers who probe before pay')
   .help('Only one --virtual invoice can be active at a time')
   .argument('[amount]', 'Amount for invoice', STRING, '0')
+  .option('--expiry <expires_in>', 'Hours the invoices expires in', INT)
```
Comment (alexbosworth) [inline bos:1065]:
```suggestion
  .option('--hours <expires_in>', 'Hours the invoices expires in', INT)
```

What about change this to make it more direct what input to give it

File: offchain/create_invoice.js [hunk 2]
```diff
@@ -112,6 +114,16 @@ module.exports = (args, cbk) => {
...
       }],
 
+      // Calculate invoice expiry
+      expiresAt: ['validate', ({}, cbk) => {
+        // Exit early if expiry is not defined
+        if (!args.expires_in) {
+          return cbk();
+        }
+
+        return cbk(null, moment().add(args.expires_in, 'hour').toISOString());
```
Comment (alexbosworth) [inline offchain/create_invoice.js:124]: I think you could just do `new Date(Date.now() + (msPerSecond * secondPerMinute * hours)).toISOString()`, would that be enough?

File: bos (hunk 1 above)
Comment (niteshbalusu11) [reply bos:1065]: like adding --hours --minutes flags separately?

Comment (alexbosworth) [reply bos:1065]: In the future maybe, I think hours is enough for the basic functionality

**Comment by niteshbalusu11** [comment]
Fixed

File: offchain/create_invoice.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ const {parseAmount} = require('ln-accounting');
 const {parsePaymentRequest} = require('ln-service');
 const {returnResult} = require('asyncjs-util');
 const {subscribeToForwardRequests} = require('ln-service');
+const moment = require('moment');
```
Comment (alexbosworth) [inline offchain/create_invoice.js:15]: i think this isn't needed if native Date is used

Comment (niteshbalusu11) [reply offchain/create_invoice.js]: Removed

**Comment by alexbosworth** [comment]
Looks good, i'll play around with this tomorrow

**Pull request merged**